pkgs/applications/networking/znc/default.nix

```diff
@@ -7,19 +7,27 @@
 
 with stdenv.lib;
 stdenv.mkDerivation rec {
-  name = "znc-1.6.2";
+  name = "znc-1.6.3";
 
-  src = fetchurl {
-    url = "http://znc.in/releases/${name}.tar.gz";
-    sha256 = "14q5dyr5zg99hm6j6g1gilcn1zf7dskhxfpz3bnkyhy6q0kpgwgf";
+  src = fetchgit {
+    url = "https://github.com/znc/znc";
+    rev = "refs/tags/${name}";
+    sha256 = "01v6b13y34mrcdin3gb1n25fh0d7p3g7bm2m33nlfqc68580ail9";
+    fetchSubmodules = true;
   };
 
-  buildInputs = [ openssl pkgconfig ]
+  patches = [ ./sslfiles.patch ./csocket.patch ]; # https://github.com/znc/znc/pull/1192 easier SSL configuration
+
+  buildInputs = [ openssl pkgconfig autoconf automake which ]
     ++ optional withPerl perl
     ++ optional withPython python3
     ++ optional withTcl tcl
     ++ optional withCyrus cyrus_sasl;
 
+  preConfigure = ''
+    $shell autogen.sh
```

Comment:
You should put `autoreconfHook` into `nativeBuildInputs` instead. You can then also remove the dependency on `autoconf` and `automake`.